app/src/main/java/intellistart/interviewplanning/model/slot/validation/SlotValidator.kt

```diff
@@ -17,45 +20,43 @@ class SlotValidator(
     private val periodService: PeriodService
 ) {
 
-    fun validateCreating(slot: Slot, email: String) {
-        validateSlotInFuture(slot)
-        validateOverlapping(slot, email)
-    }
-
-    fun validateUpdating(slot: Slot, email: String) {
-        validateSlotIsBookingAndTheSlotExists(slot.id)
-        validateSlotInFuture(slot)
-        validateOverlapping(slot, email)
-    }
+    fun validateCreating(slot: Slot, email: String): Mono<Unit> =
+        Mono.`when`(
+            Mono.fromCallable {
+                validateSlotInFuture(slot)
+            },
+            validateOverlapping(slot, email)
+        ).thenReturn(Unit)
+
+    fun validateUpdating(slot: Slot, email: String): Mono<Unit> =
+        Mono.`when`(
+            Mono.fromCallable {
+                validateSlotInFuture(slot)
+            },
+            validateSlotIsBookingAndTheSlotExists(slot.id),
+            validateOverlapping(slot, email)
+        ).thenReturn(Unit)
 
     private fun validateSlotInFuture(slot: Slot) {
         if (LocalDate.now() > slot.period.date) {
             throw SlotException(SlotExceptionProfile.INVALID_BOUNDARIES)
         }
     }
 
-    private fun validateOverlapping(slot: Slot, email: String) {
-        val slotList = slotService.getSlotsByEmailAndDate(
-            email,
-            slot.period.date
-        )
-
-        if (slotList.isNotEmpty()) {
-            for (item in slotList) {
-                if (slot.id != item.id &&
-                    periodService.areOverlapping(slot.period, item.period)
-                ) {
-                    throw SlotException(SlotExceptionProfile.SLOT_IS_OVERLAPPING)
-                }
+    private fun validateOverlapping(slot: Slot, email: String): Mono<Unit> =
+        slotService.getSlotsByEmailAndDate(email, slot.period.date)
+            .filter {
+                slot.id != it.id && periodService.areOverlapping(slot.period, it.period)
+            }.hasElements().flatMap {
+                Unit.toMono()
+            }.switchIfEmpty { SlotException(SlotExceptionProfile.SLOT_IS_OVERLAPPING).toMono() }
```

Comment:
```suggestion
            }
            .switchIfEmptyDeferred { 
                SlotException(SlotExceptionProfile.SLOT_IS_OVERLAPPING).toMono() 
            }
            .then(Unit.toMono())
```

Reply:
I had a bug in the logic , so I'll offer my corrected version of it